simulation/validator/miner_data_handler.py

```diff
@@ -57,40 +56,24 @@ def set_reward_details(reward_details: [], validation_time: str, start_time: str
                 connection.rollback()
                 bt.logging.info(f"in set_reward_details (got an exception): {e}")
 
-    @staticmethod
-    def get_values(miner_uid: int, current_time_str: str):
-        """Retrieve the record with the longest valid interval for the given miner_id."""
-        current_time = datetime.fromisoformat(current_time_str)
-
-        best_record = None
-        max_end_time = current_time - timedelta(days=5)
-
-        with engine.connect() as connection:
-            query = select(miner_predictions.c.prediction).where(
-                miner_predictions.c.validation_time >= max_end_time,
-                miner_predictions.c.validation_time <= current_time,
-                miner_predictions.c.miner_uid == miner_uid
-            )
-            result = connection.execute(query)
-
-            # Fetch all results
-            predictions = [row.prediction for row in result]
-
-        bt.logging.info("in get_values, predictions length:" + str(len(predictions)))
-
-        # Find the record with the longest valid interval
-        for prediction in predictions:
-            if prediction is None:
-                continue
+@staticmethod
+def get_values(miner_uid: int, current_time_str: str):
+    """Retrieve the record with the longest valid interval for the given miner_id."""
+    current_time = datetime.fromisoformat(current_time_str)
+    max_end_time = current_time - timedelta(days=5)
 
-            end_time = datetime.fromisoformat(prediction[-1]["time"])
+    with engine.connect() as connection:
+        query = select(miner_predictions.c.prediction).where(
+            miner_predictions.c.validation_time >= max_end_time,
+            miner_predictions.c.validation_time <= current_time,
+            miner_predictions.c.miner_uid == miner_uid,
+        ).order_by(desc(func.cast(func.jsonb_array_elements(miner_predictions.c.prediction)[-1].op('->>')('time'), DateTime))).limit(1)
```

Comment:
@parkhomenko I'm not sure at all if this is what we want

Reply:
@Thykof @parkhomenko @SamHT Hey you all. we really like your project and love to talk more about it and how we can launch it. What are the ways to connect with you? This is my email hi@chainnews.one also you can DM me on our X accounts @MatrixOneAI and @ChainNewsOne

Reply:
no, we change the logic